src/Commands/RunScan.ts

```diff
@@ -219,4 +227,67 @@ export class RunScan implements CommandModule {
       process.exit(1);
     }
   }
+
+  private async validateParams(
+    args: Arguments,
+    scanManager: Scans
+  ): Promise<void> {
+    const projectId = args.project as string;
+    if (!projectId) {
+      logger.error(
+        'Please specify a project ID with the --project (-p) option to run a scan with entrypoints using the --entrypoint option.\n\nExamples:\nbright scan:run --project <project_id> --entrypoint\nbright scan:run --project <project_id> --entrypoint <entry_point_id1> <entry_point_id2> <entry_point_id3>'
+      );
+      process.exit(1);
+    }
+
+    const isProjectExist = await scanManager.isProjectExist(projectId);
+    if (!isProjectExist) {
+      logger.error(
+        `The specified project ID ${projectId} does not exist.\nPlease verify the project ID and try again.`
+      );
+      process.exit(1);
+    }
+
+    const entryPointIds = args.entrypoint as string[];
+    if (entryPointIds?.length) {
+      const maxAllowedEntryPoints = 2000;
+      if (entryPointIds.length > maxAllowedEntryPoints) {
+        logger.error(
+          `You have specified ${entryPointIds.length} entrypoints, maximum of ${maxAllowedEntryPoints} entrypoints allowed in a single scan.\nPlease reduce the number and try again.`
+        );
+        process.exit(1);
+      }
+
+      await this.validateEntryPointsFromSameProject(
+        projectId,
+        args.entrypoint as string[]
+      );
+    }
+  }
+
+  private async validateEntryPointsFromSameProject(
+    projectId: string,
+    entryPointIds: string[]
+  ): Promise<void> {
+    const entryPoints: RestEntryPoints = container.resolve(RestEntryPoints);
+    const invalidEntryPointIds: string[] = [];
+    for (const entryPointId of entryPointIds) {
+      const entryPointDetails = await entryPoints.getEntryPointDetails(
+        projectId,
+        entryPointId
+      );
+      if (!entryPointDetails) {
+        invalidEntryPointIds.push(entryPointId);
+      }
+    }
+
+    if (invalidEntryPointIds.length) {
+      logger.error(
+        `The following entrypoint IDs do not belong to the specified project ID ${projectId}.\nPlease ensure all entrypoint IDs belong to the specified project and try again.\n\nInvalid entrypoint IDs:\n${invalidEntryPointIds.join(
```

Comment:
This error message can be implemented on `POST /api/v1/scans` thus avoiding `O(n)` calls to BE. API endpoints can be rate limited, so doing 2000 api calls in worst case can easily fail/take very long time